src/ansys/simai/core/api/training_data.py

```diff
@@ -65,3 +67,12 @@ def compute_training_data(self, training_data_id: str) -> None:
 
     def get_training_data_subset(self, project_id: str, training_data_id: str) -> Dict[str, Any]:
         return self._get(f"projects/{project_id}/data/{training_data_id}/subset")
+
+    def put_training_data_subset(
+        self, project_id: str, training_data_id: str, subset: SubsetEnum
+    ) -> None:
+        if subset not in SubsetEnum.__members__.values():
+            raise InvalidArguments("Must be one of: Ignored, Training, Test, Validation.")
```

Comment:
IMO this is the raw API interface, so in here take in `subset: str` and don't check `if subset not in SubsetEnum.__members__.values():`, this should be done in `def assign_subset`